### PR TITLE
Fix PR #329

### PR DESCRIPTION
**Pull request by Jonghakseo**
### COMMITS

- [feat: add missing permission](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/494/commits/f488579e337e09dca1788f00c18b39d8c97088d7)
- [fix: check content file test regex](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/494/commits/a5a07e3055d046545348d1bf57e1c43135924a55)
- [feat: delete injected and add content runtime](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/494/commits/4008e695311c4b3e059f78e48f246b215b3ea02e)
- [feat: add runtime ui component](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/494/commits/300095e7cb89e51d1025a081f6e421aec6346ad8)

### VIDEO

https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/assets/53500778/8d86e93f-6b9c-4d81-aa27-dbde4364a4c3

### CODE REVIEW

File: src/pages/injectedContent/ui/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { createRoot } from 'react-dom/client';
+import App from '@pages/injectedContent/ui/app';
+import refreshOnUpdate from 'virtual:reload-on-update-in-view';
+import injectedStyle from './injected.css?inline';
+
+refreshOnUpdate('pages/injectedContent');
+
+const root = document.createElement('div');
+root.id = 'chrome-extension-boilerplate-react-vite-content-view-root';
```
Comment (PatrykKuniczak) [inline src/pages/injectedContent/ui/index.tsx:9]:
```suggestion
root.id = 'chrome-extension-boilerplate-react-vite-injected-content-view-root';
```

File: src/pages/injectedContent/ui/index.tsx
Comment (PatrykKuniczak) [inline src/pages/injectedContent/ui/index.tsx:1]: IMO the folder name should be 'injected-content' to follow the current naming convention

File: src/pages/popup/Popup.tsx [hunk 2]
```diff
@@ -9,6 +9,15 @@ import withErrorBoundary from '@src/shared/hoc/withErrorBoundary';
 const Popup = () => {
   const theme = useStorage(exampleThemeStorage);
 
+  const injectContentScript = async () => {
+    const [tab] = await chrome.tabs.query({ currentWindow: true, active: true });
+    console.log(tab.id)
```
Comment (PatrykKuniczak) [inline src/pages/popup/Popup.tsx:14]: This is for test i think
Don't forget to remove it :)

File: vite.config.ts [hunk 3]
```diff
@@ -34,8 +34,10 @@ export default defineConfig({
       input: {
         devtools: resolve(pagesDir, 'devtools', 'index.html'),
         panel: resolve(pagesDir, 'panel', 'index.html'),
+        injectedContent: resolve(pagesDir, 'injectedContent', 'index.ts'),
         contentInjected: resolve(pagesDir, 'content', 'injected', 'index.ts'),
         contentUI: resolve(pagesDir, 'content', 'ui', 'index.ts'),
+
```
Comment (PatrykKuniczak) [inline vite.config.ts:40]: This separate sth, i think you copied it from #329 
I think this blank space is unnecessary

**Review by PatrykKuniczak — CHANGES_REQUESTED** [review]

File: src/pages/injectedContent/ui/index.tsx
Comment (Jonghakseo) [reply src/pages/injectedContent/ui/index.tsx:1]: That's right, and we currently have two files named 'injected', so I think we need to do some cleanup.

File: src/pages/popup/Popup.tsx (hunk 2 above)
Comment (Jonghakseo) [reply src/pages/popup/Popup.tsx:14]: Thank you for the double check.

File: vite.config.ts (hunk 3 above)
Comment (Jonghakseo) [reply vite.config.ts:40]: Yes, that change came from manually modifying the conflicts, I'll remove it ^^

**Comment by PatrykKuniczak** [comment]
@Jonghakseo Are you finish this in the near future?

**Comment by Jonghakseo** [comment]
> @Jonghakseo Are you finish this in the near future?

Yep. I'm gonna finish this PR after current issue.

File: vite.config.ts (hunk 3 above)
Comment (Jonghakseo) [reply vite.config.ts]: [81b5d02](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/494/commits/81b5d02fd5ef65accf2cdafcc7f5b7674e01160b)

File: src/pages/injectedContent/ui/index.tsx
Comment (Jonghakseo) [reply src/pages/injectedContent/ui/index.tsx:1]: Newly named `content/runtime`.

File: src/pages/popup/Popup.tsx (hunk 2 above)
Comment (Jonghakseo) [reply src/pages/popup/Popup.tsx]: I deleted `conosle.log`

**Review by PatrykKuniczak — CHANGES_REQUESTED** [review]
If user try to inject it to the place where he can't e.g extension store page/start page, error occur:
|image

I think good idea is handle that and create e.g console.error or sth like that for our purpose, then every dev can handle it in they own way, but when they see our exception handling in code, they probably don't forget to handle that case :)

**Comment by PatrykKuniczak** [comment]
@Jonghakseo Let's slowly end this looong thread 😁

File: src/pages/content/runtime/injected.css
[the remark this answers is not in the document]
Comment (Jonghakseo) [reply src/pages/content/runtime/injected.css:1]: [`0dfe8a4` (#494)](https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/494/commits/0dfe8a41337b8e5c1f6dd06b3b021cdd42e4da9c)

**Comment by Jonghakseo** [comment]
> If user try to inject it to the place where he can't e.g extension store page/start page, error occur: 
> I think good idea is handle that and create e.g console.error or sth like that for our purpose, then every dev can handle it in they own way, but when they see our exception handling in code, they probably don't forget to handle that case :)

IMO, each error message seems to clearly indicate what the problem is, and realistically, there may not be a single cause for every error case. Therefore, I think it is best to leave error handling to the user's responsibility.

**Pull request merged**